Code/Framework/AzCore/AzCore/Debug/Profiler.h

```diff
@@ -5,310 +5,50 @@
-    }
+#   define AZ_PROFILE_SCOPE(category, ...) ::AZ::ProfileScope AZ_JOIN(azProfileScope, __LINE__){ #category, __VA_ARGS__ }
+#   define AZ_PROFILE_FUNCTION(category) AZ_PROFILE_SCOPE(category, AZ_FUNCTION_SIGNATURE)
 
+// Prefer using the scoped macros which automatically end the event (AZ_PROFILE_SCOPE/AZ_PROFILE_FUNCTION)
+#   define AZ_PROFILE_BEGIN(category, ...) ::AZ::ProfileScope::BeginRegion(#category, __VA_ARGS__)
+#   define AZ_PROFILE_END() ::AZ::ProfileScope::EndRegion()
 #endif // AZ_PROFILER_MACRO_DISABLE
 
-#ifndef AZ_PROFILE_FUNCTION
-// No other profiler has defined the performance markers AZ_PROFILE_SCOPE (and friends), fallback to a Driller implementation
-#   define AZ_INTERNAL_PROF_VERIFY_CAT(category) static_assert(category < AZ::Debug::ProfileCategory::Count, "Invalid profile category")
-#   define AZ_INTERNAL_PROF_CAT_NAME(category) AZ::Debug::ProfileCategoryNames[static_cast<AZ::u32>(category)]
-
-#   define AZ_PROFILE_FUNCTION(category) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-#   define AZ_PROFILE_FUNCTION_STALL(category) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-#   define AZ_PROFILE_FUNCTION_IDLE(category) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-
-#   define AZ_PROFILE_SCOPE(category, name) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category)); (void)(name)
-#   define AZ_PROFILE_SCOPE_STALL(category, name) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category)); (void)(name)
-#   define AZ_PROFILE_SCOPE_IDLE(category, name) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category)); (void)(name)
-
-#   define AZ_PROFILE_SCOPE_DYNAMIC(category, ...) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-#   define AZ_PROFILE_SCOPE_STALL_DYNAMIC(category, ...) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-#   define AZ_PROFILE_SCOPE_IDLE_DYNAMIC(category, ...) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); AZ_PROFILE_TIMER(AZ_INTERNAL_PROF_CAT_NAME(category))
-#endif
-
-#ifndef AZ_PROFILE_EVENT_BEGIN
-// No other profiler has defined the performance markers AZ_PROFILE_EVENT_START/END, fallback to a Driller implementation (currently empty)
-#   define AZ_PROFILE_EVENT_BEGIN(category, name) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category); (void)(name)
-#   define AZ_PROFILE_EVENT_END(category) \
-        AZ_INTERNAL_PROF_VERIFY_CAT(category)
-#endif
-
 #ifndef AZ_PROFILE_INTERVAL_START
 // No other profiler has defined the performance markers AZ_PROFILE_INTERVAL_START/END, fallback to a Driller implementation (currently empty)
```

Comment:
Do we need to mention 'Driller" in the comments anymore?

Reply:
Will address this in a future pass when we decide what to do with "datapoint" and "interval"